DEVELOPMENT.md

```diff
@@ -87,19 +90,13 @@ export KO_DOCKER_REPO='gcr.io/my-gcloud-project-id'
 
 ### Check out your fork
 
-The Go tools require that you clone the repository to the
-`src/knative.dev/serving` directory in your
-[`GOPATH`](https://github.com/golang/go/wiki/SettingGOPATH).
```

Comment:
I think some of the code generation stuff still actually relies on being in the gopath, unfortunately. At least, I've never got things like `./hack/update-codegen.sh` to generate proper injection stuff in the right location unless the project is checked out in the GOPATH :(.

cc @n3wscott @mattmoor to check I'm not missing a trick here!

Reply:
I believe we can drop all `GOPATH` mentions now in our docs now

With 1.13 module mode is turned on if there's a `go.mod` around - https://golang.org/doc/go1.13#modules
With 1.14 `-mod=vendor` isn't needed explicitly if there's a `vendor` folder

So technically with that it should be running the right codegen (from vendor) and we should be able to start removing some of older `GO*` env vars

Reply:
The exception might be setting the `PATH` to include your module install location

Reply:
I dont think the problem is finding the codegen tools, the problem is they try to output into the gopath which doesn't work nicely if your code isn't checked out there (unless this got fixed recently, it was the case last time I tried).

Reply:
Yeah you're right - below I wasn't `rm -rf pkg/client` so it didn't fail

```sh
$ docker run -ti golang:1.15
$ cd /
$ git clone https://github.com/knative/serving.git
$ cd serving
$ rm -rf pkg/client
$ ./hack/update-codegen.sh
```

Reply:
We have a trick to use in the hack dir but we have not applied it yet. So you still need to codegen in the gopath

Reply:
I think now https://github.com/knative/serving/pull/10245 (thanks @n3wscott!) has landed removing these lines is OK now \o/